core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/list/kopt/FlipSublistMove.java

```diff
@@ -0,0 +1,208 @@
+package org.optaplanner.core.impl.heuristic.selector.move.generic.list.kopt;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import org.optaplanner.core.api.score.director.ScoreDirector;
+import org.optaplanner.core.impl.domain.variable.descriptor.ListVariableDescriptor;
+import org.optaplanner.core.impl.heuristic.move.AbstractMove;
+import org.optaplanner.core.impl.score.director.InnerScoreDirector;
+
+/**
+ * Flips a sublist of a list variable, (the same thing as a 2-opt, but no shift to restore the original origin).
+ * For instance, given [0, 1, 2, 3, 4], fromIndexInclusive = 1, toIndexExclusive = 3, the list after the move
+ * would be [0, 3, 2, 1, 4]. If toIndexExclusive is before fromIndexInclusive, the flip is performed on the combined
+ * sublists [fromIndexInclusive, size) and [0, toIndexExclusive). For instance,
+ * given [0, 1, 2, 3, 4, 5, 6], fromIndexInclusive = 5, toIndexExclusive = 2, the list after the move would be
+ * [6, 5, 2, 3, 4, 1, 0] (and not [0, 6, 5, 2, 3, 4, 1]).
```

Comment:
The convention we use for comments is that every sentence, or its individual part, comes on its own line. Helps with reading comprehension. For example:

```suggestion
 * Flips a sublist of a list variable, (the same thing as a 2-opt, but no shift to restore the original origin).
 * For instance, given [0, 1, 2, 3, 4], fromIndexInclusive = 1, toIndexExclusive = 3, 
 * the list after the move would be [0, 3, 2, 1, 4]. 
 * If toIndexExclusive is before fromIndexInclusive, 
 * the flip is performed on the combined sublists [fromIndexInclusive, size) and [0, toIndexExclusive). 
 * For instance, given [0, 1, 2, 3, 4, 5, 6], fromIndexInclusive = 5, toIndexExclusive = 2, 
 * the list after the move would be [6, 5, 2, 3, 4, 1, 0] (and not [0, 6, 5, 2, 3, 4, 1]).
```

Also, where you're mentioning 2-opt, please refer to the class directly using @link.
Please apply consistently throughout the code.